main.py

```diff
@@ -1,389 +1,387 @@
 from __future__ import print_function
 
+from datapackage import push_datapackage
+from psycopg2.extensions import AsIs
+from sqlalchemy import create_engine
 from os.path import dirname, join
+from string import Template
+from textwrap import dedent
 
-import os
+import datapackage
 import tempfile
 import shutil
-import json
 import urllib
-import psycopg2
 import boto3
+import json
+import csv
+import os
+
 
+#utils
 def rpath(*args):
     return join(dirname(__file__), *args)
 
-env = json.load(open(rpath('.env.json')))
+
+def load_config(config_path):
+    '''
+    Load the regular config file
+    '''
+    template = open(config_path).read()
+    try:
+        config_str = Template(template).substitute(os.environ)
+    except KeyError as e:
+        raise ValueError(
+            "A missing environment variable: {}".format(e))
+    config = json.loads(config_str)
+
+    return config
+
+
+def is_s3_path(str):
+    return str.startswith("s3://")
+
+
+def split_s3_path(s3_address):
+    if not is_s3_path(s3_address):
+        raise ValueError("{} is not an S3 address".format(s3_address))
+    else:
+        (s3_bucket, s3_path) = s3_address[5:].split('/', 1)
+    return (s3_bucket, s3_path)
+
+
+config = load_config(rpath('config.json'))
+
+CYBERGREEN_SOURCE_ROOT = config['source_path']
+CYBERGREEN_DEST_ROOT = config['dest_path']
+REDSHIFT_ROLE_ARN = config['role_arn']
+REDSHIFT_URI = config['redshift_uri']
+RDS_URI = config['rds_uri']
+REF_DATA_URLS = [inventory['url'] for inventory in config['inventory']]
 # AWS credentials
-AWS_ACCESS_KEY = env['AWS_ACCESS_KEY']
-AWS_ACCESS_SECRET_KEY = env['AWS_ACCESS_SECRET_KEY']
-
-STAGE= env['STAGE']
-SOURCE_S3_BUCKET = env['SOURCE_S3_BUCKET']
-SOURCE_S3_KEY = join(STAGE,env['SOURCE_S3_KEY'])
-DEST_S3_BUCKET = env['DEST_S3_BUCKET']
-DEST_S3_KEY= join(STAGE,env['DEST_S3_KEY'])
-REFERENCE_KEY = join(STAGE, env['REFERENCE_KEY'])
-REDSHIFT_ROLE_ARN = env['REDSHIFT_ROLE_ARN']
-
-tablename = 'logentry'
-
-create_table_name = '''
-CREATE TABLE %s (
-   date timestamp,
-   ip   varchar(30),
-   risk int,
-   asn  bigint,
-   place varchar(3)
-   )
-''' % tablename
-
-print('Connecting ...')
-# Redshift connection
-connection = psycopg2.connect(
-    database=STAGE,
-    user=env['REDSHIFT_USER'],
-    password=env['REDSHIFT_PASSWORD'],
-    host=env['REDSHIFT_HOST'],
-    port=env['REDSHIFT_PORT']
-)
-# S3 connection
+AWS_ACCESS_KEY = config['access_key']
+AWS_ACCESS_SECRET_KEY = config['secret_key']
+# set connections
+connRedshift = create_engine(REDSHIFT_URI).connect()
+connRDS = create_engine(RDS_URI).connect()
 conns3 = boto3.resource('s3',
-	aws_access_key_id=AWS_ACCESS_KEY,
-	aws_secret_access_key=AWS_ACCESS_SECRET_KEY)
-# RDS connection
-connRDS = psycopg2.connect(
-    database=STAGE,
-    user=env['RDS_USER'],
-    password=env['RDS_PASSWORD'],
-    host=env['RDS_HOST'],
-    port=env['RDS_PORT']
-    )
+    aws_access_key_id=AWS_ACCESS_KEY,
+    aws_secret_access_key=AWS_ACCESS_SECRET_KEY
+)
 
```

Comment:
Can we move the connections inside a function? So that it should not connect on import of main.py from tests? Right now I can't launch tests because I already lost my redshift access credentails prom the past. And those should not be required to run tests.

We can do for example:
```
_connRedshift = None
def connRedshift():
    global _connRedshift
    if _connRedshift is None:
         _connRedsift = create_engine(REDSHIFT_URI).connect()
    return _connRedsift
    
```